test/extended/baremetal/helper.go

```diff
@@ -0,0 +1,198 @@
+// specified extra worker. If not found, the test will fail
+func (b *BaremetalTestHelper) getExtraWorkerData(index int) *unstructured.Unstructured {
+	ew, err := b.extraWorkersRetrieveData()
+	o.Expect(err).ToNot(o.HaveOccurred())
+
+	yamlData, ok := ew.Data[b.extraWorkerKey(index)]
+	o.Expect(ok).To(o.BeTrue(), fmt.Sprintf("unable to find data for extra worker %d", index))
+
+	jsonData, err := yaml.YAMLToJSON(yamlData)
+	o.Expect(err).ToNot(o.HaveOccurred())
+
+	var host unstructured.Unstructured
+	err = host.UnmarshalJSON(jsonData)
+	o.Expect(err).ToNot(o.HaveOccurred())
+
+	return &host
+}
+
+// GetExtraWorkerData gets baremetal host and secret for the specified extra worker.
+// If not found, the test will fail
+func (b *BaremetalTestHelper) GetExtraWorkerData(index int) (*unstructured.Unstructured, *v1.Secret) {
+	return b.getExtraWorkerData(index), b.getExtraWorkerSecretData(index)
+}
+
+// CreateExtraWorker creates a new BaremetalHost (and associated Secret).
+// If successfull, returns the newly created host and secret resources
+func (b *BaremetalTestHelper) CreateExtraWorker(host *unstructured.Unstructured, secret *v1.Secret) (*unstructured.Unstructured, *v1.Secret) {
+
+	secret, err := b.clientSet.CoreV1().Secrets("openshift-machine-api").Create(context.Background(), secret, metav1.CreateOptions{})
+	o.Expect(err).ToNot(o.HaveOccurred())
+
+	host, err = b.bmcClient.Create(context.Background(), host, metav1.CreateOptions{})
+	o.Expect(err).ToNot(o.HaveOccurred())
+
+	b.hostsCreated = append(b.hostsCreated, host)
+
+	return host, secret
+}
+
+// DeleteAllExtraWorkers deletes all the extra workers created in the current session
+func (b *BaremetalTestHelper) DeleteAllExtraWorkers() {
+	for _, host := range b.hostsCreated {
+		hostName := host.GetName()
+
+		err := b.bmcClient.Delete(context.Background(), hostName, metav1.DeleteOptions{})
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		b.waitForDeletion(host)
+	}
+}
+
+func (b *BaremetalTestHelper) waitForDeletion(host *unstructured.Unstructured) {
+	err := wait.PollImmediate(1*time.Second, 1*time.Minute, func() (done bool, err error) {
+		_, err = b.bmcClient.Get(context.TODO(), host.GetName(), metav1.GetOptions{})
+		if err != nil {
+			if errors.IsNotFound(err) {
+				return true, nil
+			}
+
+			return false, err
+		}
+		return false, nil
+	})
+	o.Expect(err).NotTo(o.HaveOccurred())
+}
+
+// WaitForProvisioningState waits for the given baremetal host to reach the specified provisioning state.
+// If successfull, returns the updated host resource, otherwise the test will fail
+func (b *BaremetalTestHelper) WaitForProvisioningState(host *unstructured.Unstructured, expectedProvisioningState string) *unstructured.Unstructured {
+
+	hostName := host.GetName()
+	var newHost *unstructured.Unstructured
+
+	g.By(fmt.Sprintf("wait until %s becomes %s", hostName, expectedProvisioningState))
+	err := wait.PollImmediate(5*time.Second, 2*time.Minute, func() (done bool, err error) {
```

Comment:
Isn't 2 minutes low for that?

Reply:
Same comment with using o.Eventually()

Reply:
@bfournie I tried to use `Eventually()` but for this case it doesn't seem a better choice than `wait.PollIImmediate`: the comparison (on the provisioning state) must be moved outside the block, and also the timeouts must be specified at the end. In addition, the error check is present but must be positional (without the possibility to specify a message if required).
So the entire logical flow appeared a little bit strange, while in the current shape is anyhow pretty linear (also, `wait.poll...` seems to be a used more frequently in the repo rather than `Eventually`)

Reply:
OK, I actually like the timeouts being specified at the end, but not being able to specify an error message seems like a limitation. Anyway, thanks for checking it out.